inst/@sym/cumsum.m

```diff
@@ -0,0 +1,80 @@
+%% Copyright (C) 2020 Tasos Papastylianou
+%%
+%% This file is part of OctSymPy.
+%%
+%% OctSymPy is free software; you can redistribute it and/or modify
+%% it under the terms of the GNU General Public License as published
+%% by the Free Software Foundation; either version 3 of the License,
+%% or (at your option) any later version.
+%%
+%% This software is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty
+%% of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <http://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @defmethod  @@sym cumsum (@var{f})
+%% @defmethodx @@sym cumsum (@var{f}, @var{n})
+%% Cumulative sum of symbolic expressions.
+%%
+%% Perform a Cumulative Sum over the rows or columns of an expression.
+%% By default, sum over the rows.  Can specify row or column sums
+%% using @var{n}.
+%%
+%% Examples:
+%% @example
+%% @group
+%% t = sym( 't' );
+%% Pi = sym( 'pi' );
+%% T = [ t; t + Pi / 2; t + Pi; t + 3 * Pi / 2; t + 2 * Pi ];
+%% C = cos(T);
+%% cumsum( C )
+%%   @result{} (sym 5×1 matrix)
+%%       ⎡     cos(t)     ⎤
+%%       ⎢                ⎥
+%%       ⎢-sin(t) + cos(t)⎥
+%%       ⎢                ⎥
+%%       ⎢    -sin(t)     ⎥
+%%       ⎢                ⎥
+%%       ⎢       0        ⎥
+%%       ⎢                ⎥
+%%       ⎣     cos(t)     ⎦
+%% @end group
+%% @end example
+%%
+%% Specify dimension along which to sum:
+%% @example
+%% @group
+%% X = repmat( sym( 'x' ), 1, 5 );
+%% cumsum( X, 2 )
+%%   @result{} (sym) [x  2⋅x  3⋅x  4⋅x  5⋅x]  (1×5 matrix)
+%% @end group
+%% @end example
+%%
+%% @seealso{@@sym/sum, @@sym/cumprod}
+%% @end defmethod
+
+
+function Y = cumsum( X, N )
+
+    if nargin < 2
+        N = 1;      % Default summation along rows
+    end
+
+    YDims = size( X );
+    NDim  = YDims(N);
+    Y     = sym( zeros( YDims ) );
+    XIdx  = substruct( '()', repmat( {':'}, 1, ndims( X ) ) );
+    YIdx  = substruct( '()', repmat( {':'}, 1, ndims( Y ) ) );
+
+    for i = 1 : NDim
+        XIdx.subs{N} = 1 : i;
+        YIdx.subs{N} = i;
+        Y = subsasgn( Y,  YIdx,  sum( subsref( X, XIdx ), N )  );
+    end
+end
```

Comment:
Ditto comments about indentation, variables, space, and unit tests for this function.